compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5109,16 +5109,22 @@ public void visit(BLangElvisExpr elvisExpr, AnalyzerData data) {
         BType actualType = symTable.semanticError;
         if (lhsType != symTable.semanticError) {
             BType referencedType = Types.getReferredType(lhsType);
-            if (referencedType.tag == TypeTags.UNION && referencedType.isNullable()) {
```

Comment:
Not directly related to this PR. However, I believe it would be better if we could refactor this logic to be similar to this
```java
BType actualType = getActualType(lhsType, elvisExpr);

private BType getActualType(BType lhsType, BLangElvisExpr elvisExpr) {
    if (lhsType == symTable.semanticError) {
        return symTable.semanticError;
    }

    BType referencedType = Types.getReferredType(lhsType);

    if (!referencedType.isNullable()) {
        dlog.error(elvisExpr.pos, DiagnosticErrorCode.OPERATOR_NOT_SUPPORTED, OperatorKind.ELVIS, lhsType);
        return symTable.semanticError;
    }

    if (referencedType.tag != TypeTags.UNION) {
        // This is a builtin union type (any, anydata, json), we use the type as is
        // since we don't have a way to represent the union - nil at the moment.
        return referencedType;
    }

    BUnionType unionType = (BUnionType) referencedType;
    LinkedHashSet<BType> memberTypes = unionType.getMemberTypes().stream()
            .filter(type -> type.tag != TypeTags.NIL)
            .collect(Collectors.toCollection(LinkedHashSet::new));

    if (memberTypes.size() == 1) {
        return memberTypes.iterator().next();
    } else {
        return BUnionType.create(null, memberTypes);
    }
}
```

Reply:
I don't see a significant benefit (other than maybe length) by doing this. The name `getActualType` is ambiguous too, right? Doesn't quite convey anything.

Reply:
Yes, the only benefit is that it makes the code easier to read. If so, we can keep the code as it is now.

Reply:
In this particular case, I don't think it makes (much of) a difference in that aspect too.

Reply:
Yes, I agree with you. I suggest this because I saw several nested if-else statements. (I tried to remove those things.)

Reply:
Ended up extracting out a method since the logic became lengthier with the latest changes.